core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -555,6 +559,26 @@ class HttpGoogleServicesDAO(val clientSecrets: GoogleClientSecrets,
     }
   }
 
+  override def isBillingAccountEnabled(billingAccount: RawlsBillingAccountName)(implicit
+    executionContext: ExecutionContext
+  ): Future[Boolean] =
+    getBillingAccountEnabled(billingAccount, getBillingServiceAccountCredential)
+
+  protected def getBillingAccountEnabled(billingAccount: RawlsBillingAccountName, credential: Credential)(implicit
+    executionContext: ExecutionContext
+  ): Future[Boolean] = {
+    implicit val service = GoogleInstrumentedService.Billing
+    val fetcher = getCloudBillingManager(credential).billingAccounts().get(billingAccount.value)
+    retryWithRecoverWhen500orGoogleError { () =>
+      val response = blocking {
+        executeGoogleRequest(fetcher)
+      }
+      response.getOpen.booleanValue()
+    } {
+      case gjre: GoogleJsonResponseException if gjre.getStatusCode / 100 == 4 => false // any 4xx error means no access
```

Comment:
So if it's a 500 or something it'll rethrow the error?

Reply:
Hmm let me take a closer look at what this is doing 👀 I think we don't want the `if gjre.getStatusCode / 100 == 4 => false` at least since a permissions error doesn't necessarily mean that the billing account is closed/deactivated